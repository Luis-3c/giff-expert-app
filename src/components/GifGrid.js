import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
/* import { getGifs } from './helpers/getGifs'; */

const GifGrid = ( {category} ) => {

    //const [images, setImages] = useState([]);
    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>
            {loading && <p className="card-grid animate__animated animate__flash">Loading</p> }
            <div className="card-grid animate__animated animate__fadeInLeft">
                    {images.map((img)=>(
                        <GifGridItem 
                            key={img.id}
                            {...img}
                        />
                    ))}
            </div>
        </>
       
    );
};

export default GifGrid;