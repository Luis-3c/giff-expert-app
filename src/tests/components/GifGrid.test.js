import React from 'react';
import {shallow} from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { render, screen } from '@testing-library/react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas del componente <GifGrid/>', () => {

    const category = "One Punch";

    test('Debe mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow( <GifGrid category=""/> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier_cosa.jpg',
            title: 'Cualquier cosa 1 '
        }, {
            id: '123',
            url: 'https://localhost/cualquier_cosa.jpg',
            title: 'Cualquier cosa 2'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        render(<GifGrid category={ category } />);
        //screen.debug();
        expect(screen.getAllByRole('img').length).toBe(gifs.length);

    })
    
    
})
