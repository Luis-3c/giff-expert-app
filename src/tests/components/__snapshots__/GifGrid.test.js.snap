// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid/> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
     
     
  </h3>
  <p
    className="card-grid animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__fadeInLeft"
  />
</Fragment>
`;
